File: src/java/infra/src/main/java/com/orders/SharedProps.java
package com.orders;

import software.amazon.awscdk.services.secretsmanager.ISecret;

public class SharedProps {
    private final String service;
    private final String env;
    private final String version;
    private final ISecret ddApiKeySecret;

    public SharedProps(String service, String env, String version, ISecret ddApiKeySecret) {
        this.service = service;
        this.env = env;
        this.version = version;
        this.ddApiKeySecret = ddApiKeySecret;
    }

    public ISecret getDdApiKeySecret() {
        return ddApiKeySecret;
    }

    public String getVersion() {
        return version;
    }

    public String getEnv() {
        return env;
    }

    public String getService() {
        return service;
    }
}

File: src/go/cdk/shared/sharedProps.go
//
// Unless explicitly stated otherwise all files in this repository are licensed
// under the Apache License Version 2.0.
// This product includes software developed at Datadog (https://www.datadoghq.com/).
// Copyright 2024 Datadog, Inc.
//

package shared

import (
	"github.com/DataDog/datadog-cdk-constructs-go/ddcdkconstruct"
	"github.com/aws/aws-cdk-go/awscdk/v2/awssecretsmanager"
)

type SharedProps struct {
	Env            string
	ServiceName    string
	Version        string
	Datadog        ddcdkconstruct.Datadog
	DDApiKeySecret awssecretsmanager.ISecret
}
